e2e/app/aou-elements/xpath-finder.ts

```diff
@@ -0,0 +1,114 @@
+import {ElementHandle, Page, WaitForSelectorOptions} from 'puppeteer';
+import TextOptions from './text-options';
+import * as widgetxpath from './xpath-defaults';
+
+/**
+ * Find a LINK or BUTTON element with a specified label.
+ * @param {string} label
+ */
+export async function findClickable(page: Page, label: string, options?: WaitForSelectorOptions): Promise<ElementHandle> {
+  const selector = widgetxpath.clickableXpath(label);
+  return page.waitForXPath(selector, options);
+}
+
+/**
+ * Find SELECT element with a specified label.
+ * @param {string} label
+ */
+export async function findSelect(page: Page, opts?: TextOptions, waitOptions?: WaitForSelectorOptions): Promise<ElementHandle> {
+  if (opts.ancestorNodeLevel === undefined) {
```

Comment:
I changed so it is required for all `TextOptions` param.

Reply:
Thanks.

Small follow-up / no action required: an alternative approach here, especially if you wanted to have 'ancestorNodeLevel' default to 2 for most methods, would be to use a default value for the parameter (nice feature from Typescript):

async function findSelect(page: Page, opts: TextOptions = { ancestorNodeLevel: 2 }, ...)

Default param values are powerful but can be overused to negative effect, so I tend to avoid them unless it's obviously the right thing to do.